---
title: "CEP Group Meeting"
date: 2023-04-19
author: Lorenzo Fabbri
institute: ISGlobal
format:
  revealjs:
    toc: false
    toc_float: true
    toc-depth: 2
    toc-location: body
    embed-resources: true
    width: 1600
    height: 800
    css: style.css
    scrollable: false
    smaller: true
bibliography: ../../../../../PhD.bib
link-citations: true
linkcolor: blue
execute:
  echo: false
  warning: false
  error: false
  message: false
editor_options: 
  chunk_output_type: console
---

```{r}
#| include: false

here::i_am("presentation.qmd")
source("../../../../../all_phd/info_phd.R")
source("../../../../../all_phd/triangulation_methods.R")
infos <- info_phd()
methods <- describe_triangulation_methods()
```

## Overview

::: {.v-center-container}
- **Aim**: to research the *short-term* effects of postnatal exposures to non-persistent EDCs on neurodevelopment and neurobehavior in childhood, and how the metabolome and the proteome might *mediate* these effects.
- **How**: by making use of the principles and criteria of triangulation [@LawlorTillingDaveySmith:2016]. We will primarily rely on the use of MTPs [@munozPopulationInterventionCausal2012; @haneuseEstimationEffectInterventions2013; @diazNonparametricCausalEffects2021] in combination with TMLE [@vanderLaanBenkeserSofrygin:2018].
- **Population**: the HELIX sub-cohort, consisting of $\text{N}=1200$ mother-child pairs.
- **Exposures**: non-persistent EDCs (phenols, phthalates, and organophosphate compounds), measured in childhood in a pool of two urine samples.
- **Outcomes**
  - Raven’s Coloured Progressive Matrices, for assessing non-verbal intelligence.
  - Computerised n-back test, for assessing working memory.
  - Attention Network Test, to provide a measure of the efficiency of three different functions of attention.
  - Child Behavior Checklist, for assessing behavioural and emotional problems.
- **Mediators**: serum and urine metabolome, plasma proteome.
:::

## The Research Questions

:::: {.columns style='display: flex !important;'}

::: {.column style='display: flex; justify-content: center; align-items: center;'}
```{mermaid}
%%| fig-cap: Simplified DAG.
%%| label: fig-diag-dag
%%| fig-align: center
flowchart LR
  exposure(chemical) --> mediator(metabolite)
  mediator --> outcome(neurodevelopment)
  exposure --> outcome
```
:::

::: {.column style='justify-content: center; align-items: center;'}
- *What are the short-term effects of childhood exposure to EDCs on the outcomes?*

A ExWAS kind of analysis with the dependent variable $y$ being the outcome, and the independent variable $x_i$ being the levels of EDC $i$.

- *What are the short-term effects of childhood exposure to EDCs on metabolites and proteins?*

A XWAS analysis with the dependent variable $m_i$ being the metabolite or protein $i$, and the independent variable $x_j$ being the levels of EDC $j$.

- *What are the short-term effects of childhood exposure to metabolites and proteins on the outcomes?*

A XWAS analysis with the dependent variable $y$ being the outcome, and the independent variable $m_i$ being the metabolite or protein $i$.
:::

::::

## Triangulation

:::: {.columns style='display: flex !important;'}

::: {.column style='display: flex; justify-content: center; align-items: center;'}
```{mermaid}
%%| fig-cap: Diagram summarizing research questions and methods.
%%| label: fig-diagram
flowchart LR
  paper[Paper 3] --> rq1("RQ1
                          neuro~chemical")
  paper          --> rq2("RQ2
                          met~chemical")
  paper          --> rq3("RQ3
                          neuro~met")
  rq1            --> rqReg("Observational study.
                            Method: Regression with MTPs.
                            Causal estimand: phi.
                            Uncertainty quantification: bootstrapping.")
  rq1            --> rq1outNeg("Outcome negative control.
                                Method: Regression with GLMs/GAMs.
                                Uncertainty quantification: bootstrapping.")
  rq1            --> rqLit("Literature search")
  
  rq2            --> rqReg
  rq2            --> rqCrossCohort("Cross-cohort comparison.
                                    Method: Regression with GLMs/GAMs.
                                    Uncertainty quantification: bootstrapping.")
  rq2            --> rqLit
  
  rq3            --> rqReg
  rq3            --> rqCrossCohort
  rq3            --> rq3mr("Mendelian Randomization.
                            Causal estimand: .")
```
:::

::: {.column .r-fit-text}
- _Chemical → outcome_
  - Multivariate analysis in observational data: application of multivariable regression to observational data ($\phi^{\Delta, i} = \mathbb{E}[Y^{d_i}] - \mathbb{E}[Y]$).
  - Outcome negative control study: aims to reproduce the same conditions as the real study, but using a different outcome not plausibly causally related to the exposure.
  - Literature search: aims to validate the obtained results with data from the published scientific literature.
- _Chemical → omic_
  - Multivariate analysis in observational data
  - Cross-cohort comparison: compares results between two or more populations in different contexts that result in different confounding structures.
  - Literature search
- _Omic → outcome_
  - Multivariate analysis in observational data
  - Cross-cohort comparison
  - Mendelian Randomization: instrumental variable is one or more genetic variant(s) that have been shown to robustly relate to the exposure.
:::

::::

## Triangulation: Details

::: {.panel-tabset}
### `outcome ~ chemical`
- **Regression analysis**. We will use TMLE in combination with SL, and a library of estimators including both simple parametric models (i.e., `glm`), and data-adaptive semi-parametric models. Specifically, we will make use of the `lmtp` R package, in combination with TMLE and SL. We hypothesize the presence of residual confounding, especially due to genetic and parental \acr{se} factors, which would result in **exaggeration** of any true causal effect.
- **Outcome negative control study**. We postulate that SE factors represent the main confounders for this association. We identified outcomes that are associated to SE factors, but that are not causally associated to the chemicals: having a car, and whether the child has their own room. We will thus test the assumptions regarding the relation between the SE factors with each outcome, and check whether these associations are in the anticipated direction.
- **Literature search**. For each `outcome ~ chemical` association, we will perform a literature search to compare the obtained results.

### `omic ~ chemical`
- **Regression analysis**. As above.
- **Cross-cohort comparison**. We will compare the regression models' results between subjects of different ethnic origins. We postulate that SE factors represent the main confounders for this association. Based on the associations between SE factors (e.g., maternal education) and the omic markers, we will compare the results of `omic ~ chemical` with the expected modified associations (e.g., weaker or stronger).
- **Literature search**. As above.

### `outcome ~ omic`
- **Regression analysis**. As above.
- **Cross-cohort comparison**. We postulate that SE factors represent the main confounders for this association. We will thus test the association between SE factors and the omics, separately for the two sub-populations (e.g., white British and Pakistani in BiB [@WrightSmallRaynor:2013]). If this association is smaller among Pakistani, if the association among white British were due to residual SE factor confounding, we would expect a **weaker** association among Pakistani.
- **MR**. We will make use, ideally, of a weighted allele score of genetic variants known to be robustly associated with the omic markers as an IV. We will use methods, including sensitivity analysis, to explore the possibility of bias due to: (i) weak instruments, and (ii) violation of the exclusion restriction criteria. We will make use of 2SMR [@DaveySmithEbrahim:2003; @PierceBurgess:2013; @DaveySmithHemani:2014]. We will thus employ results from two different types of GWASs: a exposure GWAS, using The Metabolomics GWAS Server [@SuhreShinPetersen:2011; @ShinFaumanPetersen:2014], and a outcome GWAS, using the OpenGWAS project [@10.7554/eLife.34408; @ElsworthLyonAlexander:2020].
:::

## Bibliography
