---
title: "Paper 3 - Causal Roadmap"
keywords:
  - Causal Inference
  - Epidemiology
  - Triangulation
  - Modified Treatment Policy
  - Mendelian Randomization
  - Metabolomics
  - Endocrine Disrupting Chemicals
  - Neurodevelopment
  - Neurobehavior
metadata-files:
  - ../../../all_phd/authors.yml

format:
  pdf:
    pdf-engine: pdflatex
    number-sections: true
    toc: false
    margin-left: 15mm
    margin-right: 15mm
    margin-bottom: 20mm
    margin-top: 15mm
    include-in-header:
      text: |
        \usepackage{rotating}
        \usepackage{float}
        \usepackage{typearea}

bibliography: ../../../PhD.bib
link-citations: true
linkcolor: blue

execute:
  echo: false
  warning: false
  error: false
  message: false
editor_options: 
  chunk_output_type: console

filters:
  - ../../../all_phd/_extensions/acronyms/parse-acronyms.lua
  - ../../../all_phd/_extensions/authors-block/authors-block.lua
acronyms:
  fromfile: ../../../all_phd/acronyms.yml
  insert_links: false
  insert_loa: false
  sorting: alphabetical
---

```{=latex}
\setcounter{tocdepth}{4}
\tableofcontents
```

```{r}
#| include: false
source("../../../all_phd/info_phd.R")
source("../../../all_phd/triangulation_methods.R")

here::i_am("causal_roadmap.qmd")
infos <- info_phd()
methods <- describe_triangulation_methods()
```

```{r}
# Type of adjustment set from DAG
type_mas <- "minimal"
# Type of effect to be estimated from DAG
type_effect <- "direct"

# Number of folds for Cross-Validation (CV)
cv.folds <- 10
```

\newpage

Most of the following was **copy-pasted** from published papers.

<!-- ####################################################################### -->
# Formulate the research question(s)
<!-- ####################################################################### -->

The **aim** of the present study is to research the short-term effects of postnatal exposures to non-persistent \acr{edcs} on neurodevelopment and neurobehavior in childhood, and how the metabolome and the proteome might mediate these effects. We will strengthen the estimated effects by making use of the principles and criteria of triangulation [@LawlorTillingDaveySmith:2016]. To estimate these effects, we will primarily rely on the use of \acr{mtps} [@munozPopulationInterventionCausal2012; @haneuseEstimationEffectInterventions2013; @diazNonparametricCausalEffects2021] in combination with \acr{tmle} [@vanderLaanBenkeserSofrygin:2018], to avoid relying on arbitrary parametric assumptions. We will consider both additive and multiplicative shifts of the exposures' distribution.

@fig-diagram summarizes the research questions and adopted methods to answer them. In case of regression analysis using \acr{mtps}, the causal quantity of interest, $\phi^{\Delta, i} = \mathbb{E}[Y^{d_i}] - \mathbb{E}[Y]$, will be *adapted* to the research question (i.e., different policies based on the type of exposure).

```{mermaid}
%%| fig-cap: Diagram summarizing research questions and methods.
%%| label: fig-diagram
flowchart LR
  paper[Paper 3] --> rq1("RQ1
                          neuro~chemical")
  paper          --> rq2("RQ2
                          met~chemical")
  paper          --> rq3("RQ3
                          neuro~met")
  rq1            --> rqReg("Observational study.
                            Method: Regression with MTPs.
                            Causal estimand: phi.
                            Uncertainty quantification: bootstrapping.")
  rq1            --> rq1outNeg("Outcome negative control.
                                Method: Regression with GLMs/GAMs.
                                Uncertainty quantification: bootstrapping.")
  rq1            --> rqLit("Literature search")
  
  rq2            --> rqReg
  rq2            --> rqCrossCohort("Cross-cohort comparison.
                                    Method: Regression with GLMs/GAMs.
                                    Uncertainty quantification: bootstrapping.")
  rq2            --> rqLit
  
  rq3            --> rqReg
  rq3            --> rqCrossCohort
  rq3            --> rq3mr("Mendelian Randomization.
                            Causal estimand: .")
```

The primary study **population** is based on the HELIX sub-cohort, consisting of $\text{N}=1200$ mother-child pairs from six existing European birth cohorts [@vrijheidHumanEarlylifeExposome2014]. We will replicate the obtained results in the HELIX Child Panel, consisting of $\text{N}=150$ children followed twice for one week. **Exposures** consisted of non-persistent \acr{edcs} (phenols, phthalates, and organophosphate compounds), measured in childhood in a pool of two urine samples. A single-spot blood sample, collected during the visit, was used for serum metabolomics and plasma proteomics. **Outcomes** related to childhood neurodevelopment and neurobehavior, included:

- \acr{cpm}, for assessing non-verbal intelligence [@RavenCourtRaven:1998]. It includes the following outcomes: sum of the right answers, response time in minutes.
- Computerised n-back test, for assessing working memory [@Kirchner:1958]. The recommended outcome is $d^{\prime}$.
- \acr{ant}, to provide a measure of the efficiency of three different functions of attention [@RuedaFanMcCandliss:2004]. The recommended outcome is \acr{hitrtse}, a measure of response speed consistency.
- \acr{cbcl}, for assessing behavioural and emotional problems [@Tm:2001]. The recommended outcomes are: internalizing scales-emotional problems, externalizing scales-behavioral problems, and total problems.

<!-- ####################################################################### -->
## Descriptive analyses
<!-- ####################################################################### -->

{{< include ../../../all_phd/descriptive_stats.qmd >}}

<!-- ####################################################################### -->
# Define a realistic statistical model
<!-- ####################################################################### -->

For time index $t$, let $W_t$ denote the set of potential confounders, $A_t$ the observed exposures, 
and $Y_t$ the clinical outcome of interest. For each subject $i$ and each time index $t$, 
we assume its observed data $O_{ti} = (W_{ti}, A_{ti}, Y_{ti})$ were generated by sampling from a distribution 
$\mathbb{P}_{0,t}$ compatible with the causal model specified in @eq-npsem.

<!-- ####################################################################### -->
# Specify a causal model and causal quantity of interest
<!-- ####################################################################### -->

```{r}
rq1.title <- "What are the short-term effects of childhood exposure to EDCs on the outcomes?"
rq2.title <- "What are the short-term effects of childhood exposure to EDCs on metabolites and proteins?"
rq3.title <- "What are the short-term effects of childhood *exposure* to metabolites and proteins on the outcomes?"
```

We specify the following \acr{npsem} to present the data generating process, 
including its measured confounders $W_t$, exposures $A_t$, and outcome $Y_t$, for each time index $t$:

$$
\begin{aligned}
  W_t &= f_{W{_t}} (U_{W_{t}}) \\
  A_t &= f_{A_{t}} (W_t, U_{A_{t}}) \\
  Y_t &= f_{Y_{t}} (W_t, A_t, U_{Y_{t}}), 
\end{aligned}
$$ {#eq-npsem}

where $(f_{W{_t}}, f_{A_{t}}, f_{Y_{t}})$ are the non-parametric structural equations, 
and $(U_{W_{t}}, U_{A_{t}}, U_{Y_{t}})$ are the unmeasured factors contributing the confounders, exposures, and outcome, respectively. 
The corresponding causal graph (without indexes for ease of notation) is shown in @fig-causal-graph.

```{r}
#| label: fig-causal-graph
#| fig.cap: Causal graph corresponding to the structural equations.
#| fig.height: 3.0
#| fig.width: 3.0
causal_graph <- ggdag::dagify(A ~ W + U_A, Y ~ W + A + U_Y, W ~ U_W, 
                              exposure = "A", outcome = "Y") %>%
  ggdag::tidy_dagitty()
ggdag::ggdag(causal_graph) + ggdag::theme_dag()
```

In order to test the \acr{npsem}â€™s implications (conditional independencies), we will perform simulations from the DAGs, checking assumptions and data consistency, using the `dagitty` R package.

Following the \acr{mtp} framework, we will generate counterfactual outcomes $Y^d = f_Y (W, A^d, U_Y)$ by intervening on the causal model to shift the observed exposures by some user-specified function $A^d = d(\cdot)$. We will focus on simple interventions to shift the observed exposure $A$ by an additive constant $c$ or multiplicative constant $k$: $d(A,c) = c+A$ and $d(A,k) = k \times A$, respectively.

We specify our causal parameter $\phi$ as the difference in the expected counterfactual outcome under the shifted exposures and the expected outcome under the observed exposures:

$$
\phi^{\Delta} = \mathbb{E}[Y^d] - \mathbb{E}[Y],
$$

where the expectation is over the individuals of our target population.

As **primary analyses**, we will estimate the short-term effects of childhood exposure to individual non-persistent \acr{edcs} in relation to one or more neurodevelopmental outcomes of interest. We will further explore the potential mediating role of the serum metabolome and the plasma proteome.

In the following sections, we present a brief summary of each research question.

## Research question 1: `r rq1.title`

We will estimate the short-term effects of childhood exposure to non-persistent \acr{edcs} on the outcomes of interest measured in childhood. We will perform an ExWAS kind of analysis with the dependent variable $y$ being the outcome, and the independent variable $x_i$ being the levels of \acr{edc} $i$. We will assume that all associations are confounded by the same set of confounders, independently of the \acr{edc} considered. We will further stratify the analyses by sex, to account for potential sex-specific effects of the chemicals.

```{r}
#| fig.cap: Simplified DAG for research question 1.
sdag1 <- ggdag::dagify(X_ti ~ W_t, 
                       Y ~ W_t + X_ti, 
                       exposure = "X_ti", outcome = "Y") %>%
  ggdag::tidy_dagitty() %>%
  ggdag::ggdag(text_size = 2, node_size = 10) + ggdag::theme_dag() +
  ggplot2::ggtitle("RQ1")
```

## Research question 2: `r rq2.title`

We will estimate the short-term effects of childhood exposure to non-persistent \acr{edcs} on metabolites and proteins measured in childhood. We will perform an XWAS analysis with the dependent variable $m_i$ being the metabolite or protein $i$, and the independent variable $x_j$ being the levels of \acr{edc} $j$. We will assume that all associations are confounded by the same set of confounders, independently of the \acr{edc} considered. We will further stratify the analyses by sex, to account for potential sex-specific effects of the chemicals.

## Research question 3: `r rq3.title`

We will estimate the short-term effects of childhood *exposure* to metabolites and proteins on the outcomes of interest measured in childhood. We will perform an XWAS analysis with the dependent variable $y$ being the outcome, and the independent variable $m_i$ being the metabolite or protein $i$. We will assume that all associations are confounded by the same set of confounders, independently of the biomarker considered.

<!-- ####################################################################### -->
# Identification and the statistical estimand
<!-- ####################################################################### -->

For the causal parameter $\phi^{\Delta}$, which involves a summary measure of the distribution of counterfactuals, to be identified in terms of the observed data distribution, several assumptions would be required:

* **No unmeasured confounding**. There are no unmeasured common causes of the exposure and the subsequent outcome, which we can formalize as $Y^d \perp \!\!\! \perp A|W$. In our context, this assumption would be violated if, for example, an unmeasured variable influences both the exposure levels and the clinical outcome. We cannot guarantee that this assumption holds, and therefore limit our interpretations to statistical associations rather than causal effects.
* **Positivity**. If $(a, w)$ is within the support of $A, W$, then $(d(a,c), w)$ for additive shifts and $(d(a,k), w)$ for multiplicative shifts must also be within the support of $A,W$. In practice, this means that for any given time index and set of adjustment covariates, there is a positive probability of finding a subject with the same covariate values and a exposure level matching the shifted value. We will attempt to improve plausibility of this assumption by considering small shifts, while recognizing that this approach makes our causal effect data-adaptive.
* **Independence** of subjects. This assumption also implies **no interference**: the exposure level $a$ for a given subject does not affect the outcomes of the other subjects.
* **Consistency**. If $A = a$ for any subject, then $Y(a) = Y$, and hence the full observed set of outcomes when $A^d = A$ is simply $Y(A^d) = Y$. This means that the counterfactual outcome for a subject with its observed exposure level is the observed outcome.
* **Time-ordering**. The confounders $W$ precede the exposure $A$, which also precedes the outcome $Y$.

If these identifiability assumptions held, we could specify and focus our estimation efforts on a statistical estimand that equals the wished-for causal effect. In the likely case that they are not satisfied, we could still specify and focus our estimation efforts on a statistical estimand that is as close as possible to the causal parameter. Factoring the joint distribution of the observed data $\mathbb{P}_0$ into $\mathbb{P}_0 (O) = \mathbb{P}_0(Y|A,W) \mathbb{P}_0(A|W) \mathbb{P}_0(W)$, it can be shown that the statistical estimand corresponding to expected counterfactual outcome under shift $d$, $\mathbb{E}[Y^d]$, is given by

$$
\psi_0 (A^d) = \int \mathbb{E} (Y | A=a^d, W=w) dF_{A,W}(a, w), 
$$

with $dF_{A,W}(a, w)$ as the joint density of received exposures $A$ and covariate levels $W$ being integrated over. We refer to $\psi_0 (A^d)$ as the *shift parameter*. Under no shift, the expected outcome was identified as $\psi_0 (A^d) = \mathbb{E} (Y)$. Therefore, our statistical estimand of interest, corresponding to the expected difference in the outcome under shifted and observed exposure levels, is

$$
\psi_0^{\Delta} = \psi_0 (A^d) - \psi_0 (A) = \int \mathbb{E} (Y | A=a^d, W=w)dF_{A,W} (a,w) - \mathbb{E} (Y).
$$

<!-- ####################################################################### -->
# Estimation from data and statistical inference
<!-- ####################################################################### -->

To estimate the expected outcome under the observed exposure $\psi_0 (A) = \mathbb{E} (Y)$, we will use the empirical mean outcome. For estimating the shift parameter $\psi_0 (A^d)$, we will use \acr{tmle}, which typically uses the factorization of the observed data distribution into an outcome regression $\bar{Q}(A,W) = \mathbb{E} (Y|A,W)$ and an intervention mechanism $g(A|W) = P(A|W)$. Then, an initial estimate of the outcome regression is updated by a fluctuation that is a function of the intervention mechanism. In this study, we will use the implementation of [@diazNonparametricCausalEffects2021]. Here, we summarize the algorithm for the shift parameter $\psi_0 (A^d)$. We first define the density ratio

$$
r(a,w) = \frac{g^d(a|w)}{g(a|w)}.
$$

Estimation of the conditional density $g$ is done by recasting it as a classification problem. It can be shown that

$$
r(a,w) = \frac{P(\Lambda=1|a,w)}{1-P(\Lambda=1|a,w)},
$$

with $\Lambda$ indicating whether each subject received the natural or shifted value of treatment. We refer to $P(\Lambda=1|a,w)$ as $\bar{\lambda}$. To implement estimation of $\bar{\lambda}$ and $\bar{Q}$ flexibly, we will use the ensemble machine learning algorithm \acr{sl} [@LaanPolleyHubbard:2007; @NaimiBalzer:2018]. In the following, we present the steps necessary to estimate $\psi_0 (A^d)$ and the associational parameter $\psi_0^{\Delta}$. We note that our bounded continuous outcome will be **scaled** to $[0,1]$, and that to handle missing covariates $W_{i,j}$, we will redefine the data to include $\delta_{ij}$, where $\delta_{ij}=1$ if the covariate was observed and $0$ otherwise.

1. Define and calculate the density ratio under the specified shift $\hat{r} (a_i, w_i)$ for each subject $i = 1, \dots, n$ from $\bar{\lambda}$ estimated via \acr{sl}.
1. Generate initial conditional expectations for each subject, denoted $\bar{Q}(A,W)$.
1. Fit the following logistic regression on the observed (scaled) $Y$ values, using the logit of the initial estimates as an offset
$$
\text{logit} (\bar{Q}^*(A,W)) = \text{logit} (\bar{Q}(A,W)) + \epsilon
$$
with weights $\hat{r} (a_i, w_i)$, calculating the estimated intercept $\hat{\epsilon}$, where $\bar{Q}^*(A,W)$ is now a targeted estimate of the conditional mean outcome.
1. Use the resulting $\hat{\epsilon}$ value to generate targeted predictions under a shift, $\bar{Q}^*(A^d,W)$:
$$
\bar{Q}^*(A^d,W) = \text{logit}^{-1} \left[ \text{logit} (\bar{Q} (A^d, W)) + \hat{\epsilon} \right].
$$
1. With the updated estimates $\bar{Q}^*(A^d,W)$ in hand (after unscaling), define:
$$
\hat{\psi}_{\text{tmle}} = \frac{1}{n} \sum_{i=1}^n \bar{Q}^*(A^d,W).
$$

Using the empirical mean $\hat{Y} = \frac{1}{n} \sum_{i=1}^nY_i$, to estimate $\psi_0(A) = \mathbb{E}(Y)$, we obtain an estimate of the statistical association parameter $\psi_0^{\Delta} = \psi_0(A^d) - \psi_0(A)$ with
$$
\hat{\psi}^{\Delta} = \hat{\psi}_{\text{tmle}} - \bar{Y}.
$$

The $95\%$ Wald-style confidence interval $\hat{\psi}^{\Delta} \pm 1.96 \sqrt{Var(\hat{\psi}^{\Delta})}$ is obtained from the influence curve.

Using `r I(cv.folds)`-fold \acr{cv}, \acr{sl} builds the best weighted combination of predictions from the algorithms listed in @tbl-lmtp-specs. In case of repeated measures of the exposures, an `ID` variable will be used to make sure that the `r I(cv.folds)`-fold \acr{cv} splits will keep observations from the same individuals in the same split.

<!-- ####################################################################### -->
# Triangulation: Improving causal inference in aetiological epidemiology
<!-- ####################################################################### -->

From [@LawlorTillingDaveySmith:2016]:

> *Triangulation is the practice of obtaining more reliable answers to research questions through integrating results from several different approaches, where each approach has different key sources of potential bias that are unrelated to each other. With respect to causal questions in aetiological epidemiology, if the results of different approaches all point to the same conclusion, this strengthens confidence in the finding. This is particularly the case when the key sources of bias of some of the approaches would predict that findings would point in opposite directions if they were due to such biases.*

@tbl-triangulation-paper3 summarizes the approaches that we will use to perform triangulation, based on [@LawlorTillingDaveySmith:2016].

{{< include ../../../all_phd/tab_triangulation_p3.qmd >}}

For each research question, and each approach, we will perform the following analyses:

- `outcome ~ chemical`
  - **Regression analysis**. We will use \acr{tmle} in combination with \acr{sl}, and a library of estimators including both simple parametric models (i.e., `glm`), and data-adaptive semi-parametric models. Specifically, we will make use of the `lmtp` R package, in combination with \acr{tmle} and \acr{sl}. We will also compare the results obtained with \acr{tmle} with those obtained with a \acr{sdre}. @tbl-lmtp-specs provides detailed information about the specifications used for `lmtp`. \textcolor{red}{TODO: specify the treatment and reference policies.} The models will be adjusted for relevant confounders, as identified with `dagitty`. We hypothesize the presence of residual confounding, especially due to genetic and parental \acr{se} factors, which would result in **exaggeration** of any true causal effect.
  - **Outcome negative control study**. We will try to identify outcomes in the same cohorts that would be affected by confounders that are relevant in the association of \acr{edcs} exposure to the clinical outcome of interest, but for which a biological/causal effect of the exposures is unlikely. We postulate that \acr{se} factors represent the main confounders for this association. We identified outcomes that are associated to \acr{se} factors, but that are not causally associated to the chemicals: having a car, and whether the child has their own room. \textcolor{red}{A possible issue is represented by the fact that these outcomes are measured on different scales.} We hypothesized that a higher \acr{se} position is positively associated to, e.g., having a car, and negatively associated to the chemicals' concentrations. This would result in a (confounded) **inverse** relationship. We will thus test the assumptions regarding the relation between the \acr{se} factors with each outcome, and check whether these associations are in the anticipated direction.
  - **Literature search**. For each `outcome ~ chemical` association, we will perform a literature search to compare the obtained results. More precisely, we will adopt the following strategies:
    - PubChem [@KimChenCheng:2023]. We will search each chemical, and report the results from the sections *Associated Disorders and Diseases* and *Chemical-Disease Co-Occurrences in Literature*. We will report the total number of hits for each section, the number of hits relevant for our outcome(s), and the number of articles for each hit.
    - The \acr{ctd} [@DavisWiegersJohnson:2023]. We will search each chemical, and report the results from the section *Diseases*. We will specifically look for *curated* associations. We will report the total number of hits, the number of hits relevant for our outcome(s), and the detailed information for each of these.
- `omic ~ chemical`
  - **Regression analysis**. As above.
  - **Cross-cohort comparison**. We will compare the regression models' results between subjects of different ethnic origins. We postulate that \acr{se} factors represent the main confounders for this association. Based on the associations between \acr{se} factors (e.g., maternal education) and the omic markers, we will compare the results of `omic ~ chemical` with the expected modified associations (e.g., weaker or stronger).
  - **Literature search**. As above.
- `outcome ~ omic`
  - **Regression analysis**. As above.
  - **Cross-cohort comparison**. We postulate that \acr{se} factors represent the main confounders for this association. We will thus test the association between \acr{se} factors and the omics, separately for the two sub-populations (e.g., white British and Pakistani in BiB [@WrightSmallRaynor:2013]). If this association is smaller among Pakistani, if the association among white British were due to residual \acr{se} factor confounding, we would expect a **weaker** association among Pakistani.
  - **\acr{mr}**. A simplified \acr{dag} is shown in @fig-dag-mr. We will make use, ideally, of a weighted allele score of genetic variants known to be robustly associated with the omic markers as an \acr{iv}. We will use methods, including sensitivity analysis, to explore the possibility of bias due to: (i) weak instruments, and (ii) violation of the exclusion restriction criteria. Based on these, we will postulate whether the obtained results might be biased towards the **null** or not. By using \acr{mr}, we assume that this approach tests *exposure* to omic markers across the whole of childhood. Specifically, we will perform the following analyses:
    - We will use the *standard* \acr{ivw} [@BurgessButterworthThompson:2013; @BowdenDelGrecoMMinelli:2019] method for primary \acr{mr} analyses.
    - We will use MR-Egger [@BowdenDaveySmithBurgess:2015], weighted median [@BowdenSmithHaycock:2016], and MR-PRESSO [@VerbanckChenNeale:2018] for sensitivity analyses. For *secondary* sensitivity analyses, we will use: Steiger filtering and/or bi-directional \acr{mr}, Cochran's Q-statistics (to check for heterogeneity), and the Egger intercept (to check for horizontal pleiotropy [^1]) \textcolor{red}{(citations needed)}.
    
    [^1]: In order to address the issue of horizontal pleiotropy, and to study the joint effects of multiple traits, another possibility is represented by the use of univariable \acr{mr} for screening purposes, followed by multivariable (with clumping for \acr{ld}) \acr{mr} [@DecinaHopkinsBowden:2023].
  
    We will make use of \acr{2smr} [@DaveySmithEbrahim:2003; @PierceBurgess:2013; @DaveySmithHemani:2014]. We will thus employ results from two different types of \acr{gwass}: a exposure \acr{gwas}, using The Metabolomics GWAS Server [@SuhreShinPetersen:2011; @ShinFaumanPetersen:2014], and a outcome \acr{gwas}, using the OpenGWAS project [@10.7554/eLife.34408; @ElsworthLyonAlexander:2020]. The effects of the selected genetic variants will then be harmonized. We will fit simple regression models to examine whether the selected \acr{ivs} predict the relevant nodes of the \acr{dag} (i.e., exposures and outcomes), using the F-statistics. Furthermore, we will fit simple regression models of the *form* `confounder ~ SNP`, adjusted for the *top* genetic \acr{pcs}, to test whether the \acr{snps} are separately associated with the confounders. As done in previous works, we will standardized all the considered outcomes. The results (both of the main and the sensitivity analyses) will be presented by means of forest plots, for the *significant* associations.
    
    In the case of \acr{1smr}, we will make use of the following criteria to select the \acr{snps} for both the exposures and the outcomes:
      - We will favour *cis*-variants.
      - We will *a priori* define a p-value threshold.
      - We will perform \acr{ld} clumping of the genetic variants.
      - We will *a priori* define a \acr{maf} threshold.
    Similarly, the models will be adjusted for the *top* genetic \acr{pcs}.
  
  From the metabolomics dataset, we will remove derived ratios and composite measures, and we will consider only the total lipids of each subclass [@DecinaHopkinsBowden:2023]. Among the remaining metabolomic features, we will check their genetic correlation.

```{r}
#| fig-cap: Simplified DAG for MR analyses of `outcome ~ omic` associations.
#| label: fig-dag-mr
#| fig-width: 5
#| fig-height: 4
coords <- tibble::tribble(
  ~name, ~x, ~y, 
  "neuro", 1, 0, 
  "metab", 0.5, 0, 
  "SNP", 0, 0, 
  "C", 0.75, 0.1
)
ggdag::dagify(
  neuro ~ metab + C, 
  metab ~ SNP + C, 
  exposure = "metab", 
  outcome = "neuro", 
  coords = coords
) %>%
  ggdag::ggdag(., layout = "circle") +
  ggdag::geom_dag_point(col = "gray") +
  ggdag::geom_dag_edges() +
  ggdag::geom_dag_text(col = "black") +
  ggplot2::theme(axis.title = ggplot2::element_blank(), 
                 axis.ticks = ggplot2::element_blank(), 
                 axis.text = ggplot2::element_blank())
```

## Mendelian Randomization: Assumptions

\acr{mr} requires three \acr{iv} assumptions (taken from [Wikipedia](https://en.wikipedia.org/wiki/Mendelian_randomization)):

- The genetic variant(s) used as instruments for the exposure is associated with the exposure (*relevance* assumption).
- There are no confounders of the association between the genetic variant(s) and the outcome (*independence* or *exchangeability* assumption).
- There is no pathway between the genetic variant(s) and the outcome, other than through the exposure (*exclusion restriction* or *no horizontal pleiotropy* assumption).

<!-- ####################################################################### -->
# Interpretation and sensitivity analyses to inform a substantive conclusion
<!-- ####################################################################### -->

## Reliable Causal Inference

{{< include ../../../all_phd/app-rel-ci.qmd >}}

## Reliable Statistical Inference

{{< include ../../../all_phd/app-rel-si.qmd >}}

## Replicability

{{< include ../../../all_phd/app-replicability.qmd >}}

<!-- ####################################################################### -->
# Appendix
<!-- ####################################################################### -->

<!-- ####################################################################### -->
## Data Checking
<!-- ####################################################################### -->

{{< include ../../../all_phd/app-data-checking.qmd >}}

<!-- ####################################################################### -->
## Checklist A: Replicability
<!-- ####################################################################### -->

Table with random seed, names, version numbers, and description of the main software packages.

| **Name** | **Version** | **Description** |
|:---------|:------------|:----------------|
| Random seed | NA | Will be set to $X$ |
| `R` | `r I(paste0(R.version$major, ".", R.version$minor))` | Statistical programming environment |
| `tlverse` | `r packageVersion("tlverse")` | Targeted Learning software ecosystem in R |
| `lmtp` | `r packageVersion("lmtp")` | Non-parametric Causal Effects of Feasible Interventions Based on Modified Treatment Policies |
| `sl3` | `r packageVersion("sl3")` | Super Learner ensemble machine learning |
| `labelled` | `r packageVersion("labelled")` | Manipulate metadata |
| `dlookr` | `r packageVersion("dlookr")` | Diagnose, explore, and transform data |
| `Hmisc` | `r packageVersion("Hmisc")` | Harrell Miscellaneous |
| `dagitty` | `r packageVersion("dagitty")` | Graphical Analysis of Structural Causal Models |
| `qreport` | `r packageVersion("qreport")` | Facilitating the composition of Quarto reports |
| ... | ... | ... |

## Checklist B1: `lmtp` package specifications

Table providing values for all non-data arguments, and brief rationale when departing from the default specification.

\textcolor{red}{Is it possible to include models from the `brms` R package?}

| **Argument** | **Setting** | **Default** (Y/N) | **Comment** |
|:-------------|:------------|:------------------|:------------|
| `learners_outcome` | `SL.mean` | N | |
| `learners_outcome` | `SL.glm` | Y | |
| `learners_outcome` | `SL.gam` | N | |
| `learners_outcome` | `SL.hal9001` | N | |
| `learners_outcome` | `SL.bart` | N | |
| `learners_trt` | `SL.mean` | N | |
| `learners_trt` | `SL.glm` | Y | |
| `learners_trt` | `SL.gam` | N | |
| `learners_trt` | `SL.hal9001` | N | |
| `learners_trt` | `SL.bart` | N | |
| `folds` | 10 | Y | The number of folds to be used for cross-fitting |
| `.learners_outcome_folds` | 10 | Y | The number of \acr{cv} folds for `learners_outcome` |
| `.learners_trt_folds` | 10 | Y | The number of \acr{cv} folds for `learners_trt` |

: Specifications for the `lmtp` R package. {#tbl-lmtp-specs}

<!-- ####################################################################### -->
## Gene-Environment Interactions
<!-- ####################################################################### -->

Genetic polymorphisms relevant for the **chemicals**:

* \acr{opps}:
  * The \acr{snps} of the *PON1* (paraoxonase 1) gene, known to influence the ability to metabolize and detoxify \acr{opps}: `rs854560` and `rs662`. [Reference](https://doi.org/10.1289/ehp.1408976).
  * The \acr{snps} of the genes coding for \acr{opps}-metabolizing enzymes: `rs1803274`, `rs1799807`, `rs662`, `rs854560`, `rs705379`, `rs4244285`, `rs3892097`, and `rs2177180`. [Reference](https://doi.org/10.1016/j.chemosphere.2015.07.079).
* Phenols:
  * The \acr{snp} of the *UGT2B15* gene: `rs1902023`. [Reference](https://doi.org/10.1016/j.envint.2020.106172).
* Phthalates:
  *  The \acr{snps} of the cytochrome P450 (*CYPs*) and UDP-glucuronosyltransferase (*UGTs*) genes: `rs1057910`, `rs1799853`, `rs4244285`, `rs12248560`, `rs3892097`, `rs1902023`, and `rs11692021`. [Reference](https://doi.org/10.1016/j.envint.2021.107046).
  * The \acr{snps} of the *PPAR*$\alpha$ gene: `rs75525202`, `rs4823902`, `rs4253690`, `rs12330015`, `rs5766698`, `rs115250492`, and `rs1800246`. [Reference](https://doi.org/10.3389/fgene.2022.855544). Not sure whether these \acr{snps} should be considered in the list below.

Genetic polymorphisms relevant for the **outcomes**:

<!-- ####################################################################### -->
## DAGs
<!-- ####################################################################### -->

```{r}
source("../DAGs/dag_v2.R")

dags <- dags()
```

<!-- ####################################################################### -->
### Research question 1: `r rq1.title`
<!-- ####################################################################### -->

```{r}
#| fig-cap: DAG for research question 1.
#| fig-align: center
#| fig-width: 30
#| fig-height: 40
#| out-extra: "keepaspectratio=true"
dag_rq <- dags$chem_to_out
myphd::visualize_dag(dag_rq)
adj_sets <- dagitty::adjustmentSets(x = dag_rq, 
                                    type = type_mas, effect = type_effect)
```

There are a total of `r length(adj_sets)` `r I(type_mas)` sufficient adjustment sets for estimating the `r I(type_effect)` effect. The first one is:

```{r}
adj_sets[[1]]
```

<!-- ####################################################################### -->
### Research question 2: `r rq2.title`
<!-- ####################################################################### -->

```{r}
#| fig-cap: DAG for research question 2.
#| fig-align: center
#| fig-width: 30
#| fig-height: 40
#| out-extra: "keepaspectratio=true"
```

There are a total of `r length(adj_sets)` `r I(type_mas)` sufficient adjustment sets for estimating the `r I(type_effect)` effect. The first one is:

```{r}
```

<!-- ####################################################################### -->
### Research question 3: `r rq3.title`
<!-- ####################################################################### -->

```{r}
#| fig-cap: DAG for research question 3.
#| fig-align: center
#| fig-width: 30
#| fig-height: 40
#| out-extra: "keepaspectratio=true"
dag_rq <- dags$marker_to_out
myphd::visualize_dag(dag_rq)
adj_sets <- dagitty::adjustmentSets(x = dag_rq, 
                                    type = type_mas, effect = type_effect)
```

There are a total of `r length(adj_sets)` `r I(type_mas)` sufficient adjustment sets for estimating the `r I(type_effect)` effect. The first one is:

```{r}
adj_sets[[1]]
```

<!-- ####################################################################### -->
# References {.unnumbered}
<!-- ####################################################################### -->

::: {#refs}
:::
